dune/grid/CpGrid.hpp

```diff
@@ -564,6 +571,27 @@ namespace Dune
             return 0;
         }
 
+        /// \brief returns the number of boundary segments within the macro grid
+        unsigned int numBoundarySegments() const
```

Comment:
If this is an issue we can store this number, since the grid is fixed anyway. For now I just want this to be a correct implementation.